src/ophyd_async/core/_readable.py

```diff
@@ -204,6 +205,14 @@ def add_readables(
             `StandardReadableFormat` documentation
         """
 
+        def is_signalr(device: Device) -> TypeGuard[SignalR]:
+            return isinstance(device, SignalR)
+
+        def assert_device_is_signalr(device: Device) -> SignalR:
+            if not is_signalr(device):
+                raise TypeError(f"{device} is not a SignalR")
+            return device
```

Comment:
Could this be written more succinctly?
```suggestion
        def assert_device_is_signalr(device: Device) -> SignalR:
            if not isinstance(device, SignalR):
                raise TypeError(f"{device} is not a SignalR")
            return device
```

Reply:
fixed: ab110f6d125f7d6b6ef714f02b4e70d6809b89d4